src/EditorFeatures/CSharpTest/UseRecursivePatterns/UseRecursivePatternsTests.cs

```diff
@@ -0,0 +1,239 @@
+            {
+                Name: string.Empty, ContainingNamespace: null
+            }
+        };
+    }
+}");
+        }
+
+        [Fact]
+        public async Task Test3()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    bool M()
+    {
+        return type.Name == ""ValueTuple"" &&
+            type.ContainingSymbol is var declContainer &&
+            declContainer.Kind == SymbolKind.Namespace [||]&&
+            declContainer.Name == ""System"";
+    }
+}",
+@"class C
+{
+    bool M()
+    {
+        return type is
+        {
+            Name: ""ValueTuple"", ContainingSymbol:
+            {
+                Name: ""System"", Kind: SymbolKind.Namespace
+            }
+
+            declContainer
+        };
+    }
+}");
+        }
+
+        [Fact]
+        public async Task Test4()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    bool M()
+    {
+        return type.Name == ""ValueTuple"" &&
+            type.ContainingSymbol is SomeType declContainer &&
+            declContainer.Kind == SymbolKind.Namespace [||]&&
+            declContainer.Name == ""System"";
+    }
+}",
+@"class C
+{
+    bool M()
+    {
+        return type is
+        {
+            Name: ""ValueTuple"", ContainingSymbol: SomeType
+            {
+                Name: ""System"", Kind: SymbolKind.Namespace
+            }
+
+            declContainer
+        };
+    }
+}");
+        }
+
+        [Fact]
+        public async Task Test5()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    bool M()
+    {
+        return type.Name == ""ValueTuple"" [||]&& type.IsStruct;
+    }
+}",
+@"class C
+{
+    bool M()
+    {
+        return type is
+        {
+            Name: ""ValueTuple"", IsStruct: true
+        };
+    }
+}");
+        }
+
+        [Fact]
+        public async Task Test6()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    bool M()
+    {
+        return type.Name == ""ValueTuple"" [||]&& !type.IsClass;
+    }
+}",
+@"class C
+{
+    bool M()
+    {
+        return type is
+        {
+            Name: ""ValueTuple"", IsClass: false
+        };
+    }
+}");
+        }
+
+        [Fact]
+        public async Task Test7()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    bool M()
+    {
+        return type.ContainingSymbol is var declContainer &&
+            declContainer.Kind == SymbolKind.Namespace &&
+            declContainer.Name == ""System"" [||]&&
+            (declContainer.ContainingSymbol as NamespaceSymbol)?.IsGlobalNamespace == true;
+    }
+}",
+@"class C
+{
+    bool M()
+    {
+        return type is
+        {
+            ContainingSymbol:
+            {
+                ContainingSymbol: NamespaceSymbol
+                {
+                    IsGlobalNamespace: true
+                }
+
+                , Name: ""System"", Kind: SymbolKind.Namespace
```

Comment:
this formatting seems off.

Reply:
this is the default formatting from SyntaxFactory, is this in the stable form or it can change? 
either way we probably need to do a formatting pass for the feature.

Reply:
To be pednatic.  SyntaxFactory does not formatting.  It just adds elastic trivia to things.  So this is how the formatting engine is handling elastic formatting here currently.

1. we probably need to make hte formatter better here.
2. even if the formatter is better, this feature will likely need specialized trivia handling to try to do a good job for this sort of conversion.